rad/chain.rad

```diff
@@ -97,23 +97,22 @@
 ;; enter-chain & enter-remote-chain
 
 (def eval__ eval)
-(def-rec eval
+(def eval
   (fn [expr env]
       ;; The horror, the horror
       (if (list? expr)
           (if (> (length expr) 0)
               (cond
                 (eq? (head expr) :enter-chain) (do
-                                                 (def arg_ (nth 1 expr))
-                                                 (def arg (head (eval arg_ env)))
+                                                 (def chain (head (eval__ (nth 1 expr) env)))
                                                  (def mod-state
                                                    (over (.. (@ :env) (@ 'eval))
                                                          (fn [x] (add-quit env x))
-                                                         (lookup :state arg)))
+                                                         (lookup :state chain)))
                                                  (list :ok mod-state))
                 (eq? (head expr) :enter-remote-chain) (do
                                                         (def arg_ (nth 1 expr))
-                                                        (def arg (head (eval arg_ env)))
+                                                        (def chain (head (eval__ (nth 1 expr) env)))
```

Comment:
Any reason not to use `arg_` instead of `(nth 1 expr)`? (Or to not remove `arg_` as above)?

Reply:
No. I fixed it